src/index.test.js

```diff
@@ -0,0 +1,122 @@
+const Index = require('./index');
+const core = require('@actions/core');
+const fs = require('fs');
+
+beforeEach(() => 
+{
+    // setup mock
+    jest.mock('@actions/core');
+    jest.spyOn(core, 'setFailed');
+    jest.mock('fs');
+    jest.spyOn(fs, 'readFileSync');
+});
+
+afterEach(() => 
+{
+    jest.clearAllMocks();
+});
+
+test('testGetProjectVersionFromMavenFile', async() => 
+{
+    var result = Index.getProjectVersionFromMavenFile('<project><version>1.0.0</version></project>');
+    expect(result).toBe('1.0.0');
+});
+
+test('testGetProjectVersionFromPackageJsonFile', async() => 
+{
+    var result = Index.getProjectVersionFromPackageJsonFile('{"version":"1.0.0"}');
+    expect(result).toBe('1.0.0');
+});
+
+test('testGetProjectVersionWithMavenFile', async() => 
+{
+    var result = Index.getProjectVersion('<project><version>1.0.0</version></project>', 'xml');
+    expect(result).toBe('1.0.0');
+});
+
+test('testGetProjectVersionWithPackageJsonFile', async() => 
+{
+    var result = Index.getProjectVersion('{"version":"1.0.0"}', 'json');
+    expect(result).toBe('1.0.0');
+});
+
+test('testGetProjectVersionWithTxtFile', async() => 
+{
+    var result = Index.getProjectVersion('1.0.0', 'txt');
+    expect(result).toBe('1.0.0');
+});
+
+test('testGetProjectVersionWithUnsupportedFile', async() => 
+{
+    var result = Index.getProjectVersion('1.0.0', 'md');
+    expect(result).toBe(undefined);
```

Comment:
Fehlt hier noch zusätzlich die Überprüfung auf `core.setFailed(fileType + ' is not supported!');`?